packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -30,6 +30,17 @@ class PoolAlreadyConnected extends BaseError {
 
 const log = createLogger('xo:xo-mixins:xen-servers')
 
+// Servers life cycle (S: server, X: xapi)
+//
+// Legend:
+// s: server
+// x: xapi
+// cacheX: the server's xapi is present in the cache(_xapis)
+// cacheP: the server's pool is present in the cache (_serverIdsByPool)
+//
+// sRegistered[] ─ sDisconnected[] ─ sxConnecting[cacheX] ─ xConnected[cacheX] -> !cacheP ----> sConnected[cacheX, cacheP]
+//                                                                             -> cacheP ---> xDisconnected[] ─ sDisconnected[]
+//
```

Comment:
I don't understand this explanation.

Also you should not have to add documentation in the PR itself, please move your explanation to the code.

Reply:
Example:

```js
// - server is disconnected:
//    - `_xapis[server.id]` is `undefined`
//    - `connectXenServer`():
//       1. do foo step
//       2. server is now connecting
//       3. do bar step
//       4. server is now connected
```